### ISSUE TITLE
[Bug]: null property breaks `get` for whole feature

### ISSUE THREAD

**Issue by KiwiKilian** [rnmapbox/maps#2613]
### Mapbox Implementation

Mapbox

### Mapbox Version

default

### Platform

iOS

### `@rnmapbox/maps` version

10.0.0-beta.69

### Standalone component to reproduce

```tsx
import { Camera, CircleLayer, MapView, ShapeSource } from '@rnmapbox/maps';
import { FeatureCollection } from 'geojson';
import React from 'react';

const styles = {
  mapView: { flex: 1 },
};

const features: FeatureCollection = {
  type: 'FeatureCollection',
  features: [
    {
      type: 'Feature',
      properties: {
        color: 'purple',
        otherValue: undefined,
      },
      geometry: {
        type: 'Point',
        coordinates: [-74.0047, 40.71427],
      },
    },
    {
      type: 'Feature',
      properties: {
        color: 'green',
        otherValue: null,
      },
      geometry: {
        type: 'Point',
        coordinates: [-74.00597, 40.71427],
      },
    },
  ],
};

const BugReportExample = () => {
  return (
    <>
      <MapView style={styles.mapView}>
        <Camera centerCoordinate={[-74.00597, 40.71427]} zoomLevel={14} />
        <ShapeSource id={'shape-source-id-0'} shape={features}>
          <CircleLayer
            id={'circle-layer'}
            style={{
              circleRadius: 16,
              circleColor: ['get', 'color'],
            }}
          />
        </ShapeSource>
      </MapView>
    </>
  );
};

export default BugReportExample;
```

### Observed behavior and steps to reproduce

When a feature has a `null` property it breaks retrieving any other property of that feature with `['get', 'somePropertyKey']`.

### Expected behavior

The black circle should be green.

![Simulator Screen Shot - iPhone SE (3rd generation) - 2023-02-16 at 13 44 19](https://user-images.githubusercontent.com/18724559/219369656-0e0963fc-23b5-4e18-87db-724eb080f491.png)

### Notes / preliminary analysis

Occured after upgrading from beta 67 to 69. So probably connected to the upgrade of the mapbox libs. Might even be upstream?

**Comment by mfazekas** [comment]
@KiwiKilian you should be able to change RNMapboxImplVersion to verify it's an upstream change. 

Also take a look at this issue.
https://github.com/rnmapbox/maps/issues/2590

I'm not sure If the answer there is correct, as this should be related to GeoJSON allowing null values or not and not the Mapbox Vector Tile specification. 

And also this: 
https://github.com/mapbox/mapbox-maps-ios/pull/1406

**Comment by KiwiKilian** [comment]
I confirmed it's upstream: https://github.com/mapbox/mapbox-maps-ios/issues/1914

**Comment by KiwiKilian** [comment]
They released a beta inlcuding a fix, so currently (until 10.13.0 is properly released and used by @rnmapbox/maps per default), you can upgrade by setting `RNMapboxMapsVersion` to `10.13.0-beta.1`.

**Comment by range-of-motion** [comment]
@KiwiKilian this just saved me a headache and a half. Ty 🙏
